### PR TITLE
Wire the mobius loop with the select country screen

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://www.pivotaltracker.com/story/show/168980808

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/di/InjectorProviderContextWrapper.kt [hunk 1]
```diff
@@ -3,23 +3,38 @@ package org.simple.clinic.di
...
-        injectors: Map<Class<*>, Any>
-    ): Context = InjectorProviderContextWrapper(base, injectors.mapKeys { it.key.name })
+        vararg injectors: Any
+    ): Context = InjectorProviderContextWrapper(base, injectors.toList())
+  }
+
+  fun find(clazz: Class<*>): Any {
+    val injector = injectors.find { clazz.isAssignableFrom(it.javaClass) }
+
+    requireNotNull(injector) {
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/di/InjectorProviderContextWrapper.kt]: Nice!

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt [hunk 2]
```diff
@@ -106,7 +105,7 @@ class SetupActivity : AppCompatActivity(), UiActions {
     val wrappedContext = baseContext
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: We are doing this everywhere, should we extract this into a separate function? Thoughts?

File: app/src/main/java/org/simple/clinic/selectcountry/SelectCountryScreen.kt [hunk 3]
```diff
@@ -86,6 +101,25 @@ class SelectCountryScreen(
     }
   }
 
+  private fun countrySelectionChanges(): Observable<CountryChosen> {
+    return supportedCountriesAdapter
+        .itemEvents
+        .ofType<Event.CountryClicked>()
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/selectcountry/SelectCountryScreen.kt]: `Event.CountryClicked`?

File: app/src/main/java/org/simple/clinic/selectcountry/adapter/SelectableCountryItem.kt [hunk 4]
```diff
@@ -5,24 +5,31 @@ import io.reactivex.subjects.Subject
...
 import org.simple.clinic.selectcountry.adapter.Event.CountryClicked
 import org.simple.clinic.widgets.ItemAdapter
 import org.simple.clinic.widgets.recyclerview.ViewHolderX
 
 data class SelectableCountryItem(
     val country: Country,
-    val isSelected: Boolean,
-    val showDivider: Boolean
+    private val isCountryChosenByUser: Boolean,
+    private val showDivider: Boolean,
```
Comment (ragunathjawahar) [inline app/src/main/java/org/simple/clinic/selectcountry/adapter/SelectableCountryItem.kt]: Is this how we are doing it? Curious if we are not using `RecyclerView.ItemDecoration` for a reason?

File: app/src/main/java/org/simple/clinic/setup/SetupActivity.kt (hunk 2 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/setup/SetupActivity.kt]: We should, I'll pick this up after finishing the feature.

File: app/src/main/java/org/simple/clinic/selectcountry/SelectCountryScreen.kt (hunk 3 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/selectcountry/SelectCountryScreen.kt]: I don't see the difference?

File: app/src/main/java/org/simple/clinic/selectcountry/adapter/SelectableCountryItem.kt (hunk 4 above)
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/selectcountry/adapter/SelectableCountryItem.kt]: I've never used the item decoration feature before so I didn't think about it. I'll revisit this after the feature is complete.

**Pull request merged**